src/coprocessor/dag/rpn_expr/impl_arithmetic.rs

```diff
@@ -314,6 +314,49 @@ impl ArithmeticOp for DecimalMultiply {
     }
 }
 
+#[derive(Debug)]
+pub struct RealMultiply;
+
+impl ArithmeticOp for RealMultiply {
+    type T = Real;
+    fn calc(lhs: &Real, rhs: &Real) -> Result<Option<Real>> {
+        let res = *lhs * *rhs;
+        if res.is_infinite() {
+            Err(Error::overflow("REAL", &format!("({} * {})", lhs, rhs)).into())
+        } else {
+            Ok(Some(res))
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct IntMultiply;
+
+impl ArithmeticOp for IntMultiply {
+    type T = Int;
+    fn calc(lhs: &Int, rhs: &Int) -> Result<Option<Int>> {
+        lhs.checked_mul(*rhs)
+            .ok_or_else(|| Error::overflow("INT", &format!("({} * {})", lhs, rhs)).into())
+            .map(Some)
+    }
+}
```

Comment:
In original `multiply_int` it decide what to do by checking the `signed`/`unsigned` type of `lhs` and `rhs`.
So should we consider using `map_int_sig(value, children, multiply_mapper)?` and implement different type of muliple functions here? @breeswish

Reply:
Yes. There is a TODO in the PR description: TODO: add multiply_mapper